### DISCUSSION TITLE
Unable to install brew: /bin/bash: line 234: bin/git: No such file or directory

### DISCUSSION THREAD

**Discussion by brucruz** [Homebrew/discussions#4518] [Getting started] [answered]
### Output of `brew config`

```shell
zsh: command not found: brew
```

### Output of `brew doctor`

```shell
zsh: command not found: brew
```

### Description of issue

I migrate the content off my old MacBook with Intel chip to my new MacBook with Apple Silicon. So I removed the Homebrew installation using:
`- /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"`
And when I tried to *reinstall* it, using:
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"`

I get the following error:
```
==> Checking for `sudo` access (which may request your password)...
Password:
==> This script will install:
/opt/homebrew/bin/brew
/opt/homebrew/share/doc/homebrew
/opt/homebrew/share/man/man1/brew.1
/opt/homebrew/share/zsh/site-functions/_brew
/opt/homebrew/etc/bash_completion.d/brew
/opt/homebrew

Press RETURN/ENTER to continue or any other key to abort:
==> /usr/bin/sudo /usr/sbin/chown -R brunocruz:admin /opt/homebrew
==> Downloading and installing Homebrew...
/bin/bash: line 234: bin/git: No such file or directory
Failed during: bin/git -c init.defaultBranch=master init --quiet
```
It seems it could be a problem with git.
If I run `git --version`, I get `git version 2.39.2 (Apple Git-143)`
And if I run `which git`, I get `/usr/bin/git`.

Maybe the problem is that git is installed in the wrong place? If so how can I fix it? Any alternative ideas?

**Comment by carlocab** [comment]
What's the output of `which -a git`?

**Reply by brucruz** [reply]
it's `/usr/bin/git`

**Reply by carlocab** [reply]
Just that? Ok. What about
```
env | grep HOMEBREW
```
?

**Reply by brucruz** [reply]
I get only this:
```
HOMEBREW_NO_ANALYTICS=1
```

**Reply by carlocab** [reply]
Strange. Ok, what about
```
/bin/bash -c 'command -v git'
```
?

**Reply by brucruz** [reply]
I get:
```
/usr/bin/git
```

**Reply by carlocab** [reply]
Ok, no idea. Try doing
```
cd /tmp
curl -OL https://raw.githubusercontent.com/Homebrew/install/master/install.sh
/bin/bash -x install.sh | tee ~/Desktop/install_log.txt
```
and then upload `install_log.txt` here. You should find it on your desktop.

**Reply by brucruz** [reply]
Sure
Here's the file
[install_log.txt](https://github.com/Homebrew/discussions/files/11518933/install_log.txt)

**Reply by carlocab** [reply]
Ah, right. The pipe will break the install script, so that suggestion doesn't work. Try again with
```
/bin/bash -x install.sh
```
then copy the full output into a `.txt` file and upload it here. (The output will be rather long.)

**Reply by brucruz** [reply]
Thanks again, you are very kind!
Here's the updated file (the log is really long)
[install_log.txt](https://github.com/Homebrew/discussions/files/11518980/install_log.txt)

**Reply by carlocab** [reply, marked as answer]
Hmm, ok. For some reason `command -v git` inside the script is returning `./bin/git`. Not really sure why that's happening. What's the output of
```
/bin/bash -c 'type command'
```
?

I suggest editing the `install.sh` script you have locally so that [this line](https://github.com/Homebrew/install/blob/12564487c52b3a4486f19a093d8f3fac31216b02/install.sh#L836)
```sh
USABLE_GIT="$(command -v git)"
```
becomes
```sh
USABLE_GIT=/usr/bin/git
```

**Reply by osalbahr** [reply]
> Hmm, ok. For some reason command -v git inside the script is returning ./bin/git. Not really sure why that's happening. What's the output of

That is very strange. I'm interested to know why. I tried to reinstall and it worked fine on both bash and zsh. It could be because I'm on Intel-based? I'd assume there's a GitHub Actions runner for Intel-based macs, I don't know about M1/M2.

**Reply by osalbahr** [reply]
@brucruz I'm curious, what does `echo $PATH` output? And `command -v git`?

**Reply by brucruz** [reply]
```
$ echo $PATH:
/Users/brunocruz/.zinit/polaris/bin:./bin:./node_modules/.bin:/Users/brunocruz/.rbenv/shims:/Users/brunocruz/.rbenv/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/go/bin:/Library/Apple/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/brunocruz/.fig/bin:/Users/brunocruz/.local/bin:/usr/local/share/android-commandlinetools/tools:/usr/local/share/android-commandlinetools/platform-tools:/usr/local/sbin:/Users/brunocruz/.pub-cache/bin
```
```
$ command -v git:
/usr/bin/git
```

But I'm not sure if it is the same as it was before, as I have installed and uninstalled a bunch of things since then.
Did it help?

**Reply by osalbahr** [reply]
Thanks!

It looks like, and this is just a guess, that the script failed because you were running `install.sh` in a directory where `./bin/git` exists, but  ran `command -v git` (and `which -a git`) in a different directory. Maybe you have `git` in `$HOME/bin/git`.

To be clear, this is not an issue on your side, but `install.sh` would need to be (re)modified to account for this possibility as the path to `git` needs to be absolute.

I typically don't see `./bin` in `PATH`, nor have a `$HOME/bin` directory. Not sure if this is something by default on newer macs or something you added.

**Reply by osalbahr** [reply]
So this is would be a simple way to reproduce the error:

```
~$ which -a git
/usr/bin/git
~$ command -v git
/usr/bin/git
~$ mkdir bin
~$ cp $(which git) bin
~$ export PATH="./bin:$PATH"
~$ which -a git
./bin/git
/usr/bin/git
~$ command -v git
./bin/git
```

It looks like Homebrew/install#775 is a fix based on the same assumption too.

**Comment by brucruz** [comment]
You're a wizard! Thanks!
It is installed... but, when I run `which brew` it returns `brew not found`
do you have an idea why?

**Reply by carlocab** [reply]
The install script should have printed instructions at the end for how to fix that. Try doing
```
(echo; echo 'eval "$(/opt/homebrew/bin/brew shellenv)"') >> ~/.zprofile
```
then restart your terminal.

**Reply by brucruz** [reply]
Fenomenal! Thank you so much!
